platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ManagedServiceOperation.java

```diff
@@ -0,0 +1,206 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+
+package org.codice.ddf.admin.configurator.impl;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.MalformedObjectNameException;
+import javax.validation.constraints.NotNull;
+
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+import org.codice.ddf.ui.admin.api.ConfigurationAdmin;
+import org.codice.ddf.ui.admin.api.ConfigurationAdminMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Transactional handler factory for creating and deleting managed services.
+ * <p>
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public abstract class ManagedServiceOperation
+        implements OperationBase, Operation<String, Map<String, Map<String, Object>>> {
+    /**
+     * Transactional handler for deleting managed services.
+     */
+    private static class DeleteHandler extends ManagedServiceOperation {
+        private final String configPid;
+
+        private final Map<String, Object> currentProperties;
+
+        private static String getFactoryPid(String configPid, ConfigurationAdminMBean cfgAdmMbean) {
+            try {
+                return cfgAdmMbean.getFactoryPid(configPid);
+            } catch (IOException e) {
+                ManagedServiceOperation.LOGGER.debug(
+                        "Error getting current configuration for pid {}",
+                        configPid,
+                        e);
+                throw new ConfiguratorException("Internal error");
+            }
+        }
+        private DeleteHandler(String configPid, ConfigurationAdmin configAdmin,
+                ConfigurationAdminMBean cfgAdmMbean) {
+            super(getFactoryPid(configPid, cfgAdmMbean), configAdmin, cfgAdmMbean);
+
+            this.configPid = configPid;
+
+            try {
+                currentProperties = cfgAdmMbean.getProperties(configPid);
+            } catch (IOException e) {
+                ManagedServiceOperation.LOGGER.debug(
+                        "Error getting current configuration for pid {}",
+                        configPid,
+                        e);
+                throw new ConfiguratorException("Internal error");
+            }
+        }
+
+        @Override
+        public String commit() throws ConfiguratorException {
+            deleteByPid(configPid);
+            return null;
+        }
+
+        @Override
+        public String rollback() throws ConfiguratorException {
```

Comment:
Instead of returning a raw string, I would suggest returning something like CreatedManagedServicePid that is backed by a String.